telemetry/appblock.go

```diff
@@ -0,0 +1,105 @@
+package telemetry
+
+import (
+	"runtime"
+	"strconv"
+	"time"
+
+	abci "github.com/cometbft/cometbft/abci/types"
+	"github.com/cosmos/cosmos-sdk/baseapp"
+	"github.com/cosmos/cosmos-sdk/telemetry"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/mempool"
+	"github.com/hashicorp/go-metrics"
+)
+
+type BlockTelemetry struct {
+	startTime            time.Time
+	initialBlockMemAlloc uint64
```

Comment:
There seems to be a lot of dependency between the prepare and recommit handlers. This is fine but I feel like you're going with a functional approach in a non-functional language.

Can we create a receiver with a local state so it's more straightforward?

Reply:
Will remove `startTime` (which is not very useful) and `initialBlockMemAlloc` (which is already handled by go default metrics), this struct will be completely removed.